### PR TITLE
Components revamp

### PR DESCRIPTION
**Pull request by bluong**
Retarget from #59 

[@bluong 1/30 23:39] EDIT - Just marking the additions so that QE knows the features that are being built in:

* The below are all changes to the Components tab on the website.
* There is now a small tutorial at the top of the page to detail how components are placed into a table layout.
* There is a table of contents above said tutorial to navigate between the component list and the tutorial.
* The component list has been completely revamped to instantiate each and every component once while providing a tooltip on hover showing the code behind the component.  Alongside this, there is now a button that allows you to copy said code onto the clipboard

### CODE REVIEW

**Comment by bluong** [comment]
Docs had a lookover for everything except the ToC

**Comment by bluong** [comment]
Docs approved

**Comment by jtlan** [comment]
Can we add code snippets showing how the tables in the example are constructed?

File: components/index.html [hunk 1]
```diff
@@ -2,25 +2,99 @@
...
+  <div class="row component-group">
+    <div class="col-md-12">
+      <h2>Plots</h2>
+      <h3>Bar Plot</h3>
+      <svg id="barPlot" height="250" title='new Plottable.Plot.Bar(xScale, yScale, true)'></svg>
+      <h3>Line Plot</h3>
+      <svg id="linePlot" height="250" title='new Plottable.Plot.Line(xScale, yScale)'></svg>
+      <h3>Area Plot</h3>
+      <svg id="areaPlot" height="250" title='new Plottable.Plot.Area(xScale, yScale)'></svg>
+      <h3>Scatter</h3>
```
Comment (jtlan) [inline components/index.html]: "Scatter Plot"

**Comment by jtlan** [comment]
Something weird going on with the Time Axis:
screen shot 2015-01-28 at 12 22 08 pm

**Comment by rcchen** [comment]
Some feedback on the new component page:
* The popovers with how to create an instance of each component is nice, but there's no way to copy that snippet to the clipboard because it disappears when you try to hover on it. Maybe add a "click to copy to clipboard" listener?
* Hovering over any grey box in the two diagrams at the top of the page show (null)
* The colors in the two diagrams at the top are nice for highlighting specific components, but the colors don't really have any inherent meaning to them. Maybe show borders on components instead, and only highlight on hover?
* Speaking of colors, they kind of depart from the rest of the theme a bit. A different palette might work out better here.

**Comment by bluong** [comment]
@rcchen I'll respond in the same bullet order
* Duly noted.  I'll look into that.
* Completely intentional.  If you were to codewise construct this table, you would need these nulls in these places for the table to work.
* Hmm interesting point.  Which would be preferable?  Having the colors already on the page on the important components, or only having the colors show up on hover?
* I stole the colors from our own Color Scale which depart from our theme pretty greatly :D If I go with the idea of color on hover.

**Comment by bluong** [comment]
I took all the feedback into account and made the following changes:

* I have added zeroclipboard as a bower dependency so that users can go the components list and copy the tooltip code by pressing the glyph next to the relevant label.

* The two chart tables now default to be all black with opacity, but if you hover over, then the appropriate cell is highlighted and the tooltip appears.  On hover out, it turns back to black with opacity.

* timeAxis bug has been fixed

* Code for how the table is built has been added.  The code surrounding how the components are set up and instantiated has been stripped out to show the exact table layout.

**Comment by bluong** [comment]
Note I still need to add InterpolatedColorScale.

**Comment by jtlan** [comment]
Looks good. My only thought is that there should be some sort of hover-over `<title>` or cursor change on the glyph so that users know they can click on it. Can we make it an actual clipboard glyph?

**Comment by jtlan** [comment]
@bluong wrote:
> I stole the colors from our own Color Scale which depart from our theme pretty greatly :D If I go with the idea of color on hover.

We should probably loop in @mud and see if we can make the site-wide palette work better somehow...

**Comment by rcchen** [comment]
Looking good! Just a couple things:
* No visual feedback to the user that anything has been copied.
* Glyphicons for some reason don't match up with the text that they're placed with even though they're both technically the same "size". Adding the following styles to the clip buttons will fix this: { font-size: 13px; margin-left: 5px; }
* Not a huge fan of the black opacity by default, makes things look really dark. Can we use white instead and keep the blue hover?

**Comment by jtlan** [comment]
Seconding @rchen's concerns.

**Comment by bluong** [comment]
Updated.  Still things to address...
@rcchen I'm liking the dark cells since the alternative I think is to have a plot with bunch of black lines to signify the borders.  My first guess is that the dark cells looks nicer (personal preference).  I'll go see how your idea works though.

@jtlan We should probably loop in @mud for the actual glyph and how to incorporate it.  I just took the most sensible glyph out of our current selection.

**Comment by bluong** [comment]
Ok got I think what you want @rcchen

**Comment by bluong** [comment]
@jtlan Can't add gridlines since the way gridlines are built they cannot be standalone.

**Comment by rcchen** [comment]
Looks like we're using Bootstrap's Glyphicons based on the download image. If that is indeed the case, may I suggest using "glyphicon glyphicon-copy"? 

image

**Comment by bluong** [comment]
I think I hit everything?  Confirm?

**Comment by jtlan** [comment]
This looks pretty boss.

**Pull request merged**